apps/web/src/components/providers/AuthProvider.tsx

```diff
@@ -39,6 +39,8 @@ export type AuthContextValue = UserState &
     redirectToSignUp: (params: { redirectURL?: string; origin?: string; anonymousId?: string }) => void;
     switchOrganization: (organizationId: string) => Promise<void>;
     reloadOrganization: () => Promise<{}>;
+    isCurrentUserAdministrator: boolean;
+    isCurrentUserEditor: boolean;
```

Comment:
I would recommend that this is where we should draw the distinction of Selectors in our code. Rather than bloating state with a number of flags (which will continue to come up), we can use functions that get the derived data based on the state. 

Then you would use some like `selectIsCurrentUserAdmin` instead.

Reply:
so if I understand correctly, the point is not to bloat the state to have this count of flags but instead supply just one function that can ask "is this user of this role"? That does seem like a good approach.

I fail to see how `selectIsCurrentUserAdmin` would help with that though, because then I am switching a flag for a function.

The way I understood the suggestion is that I can have a this function in the auth state:
```js
// signature
isCurrentUserInRole(role: string)

// call
const { isCurrentUserInRole } = useAuth();
...
isCurrentUserInRole(CLERK_ROLES.ADMINISTRATOR)

```
Am I understanding the suggestion correctly?